### PR TITLE
feature: S3C-1640 Add functional tests

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Add functional tests that use role to authenticate requests when retrying CRR operations.

* Refactor functional tests into individual test suites based on feature.
* Run the retry tests two times: once with account auth, and another time with role auth

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,6 +7,11 @@ const runServer = require('./lib/api/BackbeatServer');
 const Config = process.env.CI !== 'true' ? require('./conf/Config') :
     require('./tests/config.json');
```
Comment (jonathan-gramain) [inline index.js]: There can be a good reason to specifically load a test file here, but if at all possible we should set BACKBEAT_CONFIG_FILE env var and let Config.js parse it (would avoid duplicating special cases and allow config validation to take place).

**Review by jonathan-gramain — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (bennettbuchanan) [reply index.js]: I agree that would be cleaner. 👍I'll aim to add multiple config files each auth case.

**Review by philipyoo — APPROVED** [review]
nice! lgtm 👍

**Comment by bennettbuchanan** [comment]
@bert-e wait

**Comment by bennettbuchanan** [comment]
@bert-e reset

**Comment by bennettbuchanan** [comment]
@ironman-machine reset

**Comment by bennettbuchanan** [comment]
@bert-e reset

File: tests/functional/api/retry.js [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+const config = require('../../config.json');
+const getUrl = require('../utils/getUrl');
+const { addMembers, addManyMembers } = require('../utils/sortedSetHelpers');
+const { makeRetryPOSTRequest, getRequest, getResponseBody } =
+    require('../utils/httpHelpers');
+const fakeLogger = require('../utils/fakeLogger');
+
+const redisConfig = { host: '127.0.0.1', port: 6379 };
+const redisClient = new RedisClient(redisConfig, fakeLogger);
+const useRoleAuth = process.env.CI_AUTH_TYPE === 'role';
```
Comment (jonathan-gramain) [inline tests/functional/api/retry.js]: Now maybe it could make more sense to check the "authType" specified in the config rather than rely on an env var.

Comment (bennettbuchanan) [reply tests/functional/api/retry.js]: Good call, updated.

**Comment by bennettbuchanan** [comment]
@bert-e reset

**Comment by bennettbuchanan** [comment]
@bert-e reset

**Comment by bennettbuchanan** [comment]
@bert-e reset

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**